src/preferences/broadcastsettings.cpp

```diff
@@ -1,417 +1,135 @@
-}
-
-int BroadcastSettings::getDefaultMaximumRetries() const {
-    return kDefaultMaximumRetries;
-}
-
-QString BroadcastSettings::getMetadataCharset() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kMetadataCharset));
-}
-
-void BroadcastSettings::setMetadataCharset(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kMetadataCharset), value);
-}
-
-QString BroadcastSettings::getDefaultMetadataCharset() const {
-    return QString();
-}
-
-QString BroadcastSettings::getMetadataFormat() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kMetadataFormat), getDefaultMetadataFormat());
-}
-
-void BroadcastSettings::setMetadataFormat(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kMetadataFormat), value);
-}
-
-QString BroadcastSettings::getDefaultMetadataFormat() const {
-    return kDefaultMetadataFormat;
-}
-
-QString BroadcastSettings::getMountpoint() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kMountPoint));
-}
-
-void BroadcastSettings::setMountPoint(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kMountPoint), value);
-}
-
-QString BroadcastSettings::getDefaultMountpoint() const {
-    return QString();
-}
-
-bool BroadcastSettings::getNoDelayFirstReconnect() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kNoDelayFirstReconnect),
-            getDefaultNoDelayFirstReconnect());
-}
-
-void BroadcastSettings::setNoDelayFirstReconnect(bool value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kNoDelayFirstReconnect), value);
-}
-
-bool BroadcastSettings::getDefaultNoDelayFirstReconnect() const {
-    return kDefaultNoDelayFirstReconnect;
-}
-
-bool BroadcastSettings::getOggDynamicUpdate() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kOggDynamicUpdate),
-            getDefaultOggDynamicUpdate());
-}
-
-void BroadcastSettings::setOggDynamicUpdate(bool value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kOggDynamicUpdate), value);
-}
-
-bool BroadcastSettings::getDefaultOggDynamicUpdate() const {
-    return kDefaultOggDynamicupdate;
-}
-
-QString BroadcastSettings::getPassword() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kPassword), getDefaultPassword());
-}
-
-void BroadcastSettings::setPassword(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kPassword), value);
-}
-
-QString BroadcastSettings::getDefaultPassword() const {
-    return QString();
-}
-
-int BroadcastSettings::getPort() const {
-    // Valid port numbers are 0 .. 65535 (16 bit unsigned)
-    int port =  m_pConfig->getValue(
-            ConfigKey(kConfigKey, kPort), getDefaultPort());
-    if (port < 0 || port > 0xFFFF) {
-        return getDefaultPort();
+BroadcastSettings::~BroadcastSettings() {
+    for(BroadcastProfile* profile : m_profiles) {
+        delete profile;
```

Comment:
Currently I rewrote this with `QSharedPointer`s because I had trouble when passing references to `std::unique_ptr`s. I'll have to come back to this later.